src/analysis/normed_space/units.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2020 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+import analysis.normed_space.basic
+import analysis.specific_limits
+import tactic.noncomm_ring
+
+/-!
+# The group of units of a normed ring
+
+In this file we prove that the group of units of a complete normed ring is an open subset of the
+ring.
+-/
+
+noncomputable theory
+
+variables {α : Type*} [normed_ring α] [complete_space α]
+
+/-- In a normed ring, an element `x` of distance less than `1` from `1` is a unit.  Here we
+construct its `units` structure.  -/
+def perturbation_unit (x : α) (h : ∥x∥ < 1) : units α :=
+{ val := 1 - x,
+  inv := tsum (λ (n : ℕ), x ^ n),
+  val_inv := mul_neg_geom_series x h,
+  inv_val := geom_series_mul_neg x h }
+
+lemma unit_of_near_unit [nonzero α] (x : units α) (y : α)
```

Comment:
I'd define `units.sub (x : units α) (y : α) (h : ∥y - x∥ < ∥((x⁻¹:units α):α)∥⁻¹) : units α`.

Reply:
I refactored the file; please check whether it is what you had in mind.